### PR TITLE
FEATURE: adds stalled_topic and user_global_notice

### PR DESCRIPTION
**Pull request by jjaffeux**
= STALLED_TOPIC =

Stalled topic is a new trigger which will trigger when a topic owner has not posted a reply in the topic for a long time.

= USER GLOBAL NOTICE =

A new script which allows to display a global notice to a specified user.

### CODE REVIEW

**Comment by jjaffeux** [comment]
Screenshot 2021-11-19 at 16 54 04

File: app/jobs/scheduled/stalled_topic_tracker.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+          stalled_after = fields.dig('stalled_after', 'value')
+          stalled_duration = ISO8601::Duration.new(stalled_after).to_seconds
+          stalled_date = stalled_duration.seconds.ago
+          categories = fields.dig('categories', 'value')
+          tags = fields.dig('tags', 'value')
+
+          StalledTopicFinder.call(
+            stalled_date,
+            categories: categories || nil,
+            tags: tags || nil
```
Comment (ZogStriP) [inline app/jobs/scheduled/stalled_topic_tracker.rb:23]: Nitpicking, but why do you need the `|| nil`? If a key is not found in `fields`, then `dig` will automatically return `nil`, right?

```ruby
> {}.dig("foo", "bar")
 => nil 
```

File: app/lib/discourse_automation/scriptable.rb [hunk 2]
```diff
@@ -24,6 +24,7 @@ def initialize(name)
       @placeholders = [:site_title]
       @triggerables = (@@plugin_triggerables[name&.to_sym] || [])
       @script = proc {}
+      @on_reset = proc {}
```
Comment (ZogStriP) [inline app/lib/discourse_automation/scriptable.rb:27]: 🤩

**Review by ZogStriP — APPROVED** [review]
LGTM 👍 Can't wait to try this out 🙌

**Pull request merged**